src/main/plugins/control-panel-plugin/control-panel-plugin.ts

```diff
@@ -0,0 +1,91 @@
+import { SearchResultItem } from "../../../common/search-result-item";
+import { PluginType } from "../../plugin-type";
+import { IconType } from "../../../common/icon/icon-type";
+import { TranslationSet } from "../../../common/translation/translation-set";
+import { UserConfigOptions } from "../../../common/config/user-config-options";
+import { SearchPlugin } from "../../search-plugin";
+import { ControlPanelOptions } from "../../../common/config/control-panel-options";
+import { ControlPanelItem } from "./control-panel-item";
+import { defaultControlPanelIcon } from "../../../common/icon/default-icons";
+import { ControlPanelItemsRetriever } from "./control-panel-items-retriever";
+import { Logger } from "../../../common/logger/logger";
+import * as Powershell from "node-powershell";
+
+export class ControlPanelPlugin implements SearchPlugin {
+    public pluginType = PluginType.ControlPanel;
+    private controlPanelItems: ControlPanelItem[];
+    private config: ControlPanelOptions;
+    private readonly logger: Logger;
+
+    constructor(config: ControlPanelOptions, logger: Logger) {
+        this.config = config;
+        this.logger = logger;
+        this.controlPanelItems = [];
+    }
+
+    public isEnabled(): boolean {
+        return this.config.isEnabled;
+    }
+
+    public execute(searchResultItem: SearchResultItem, privileged: boolean): Promise<void> {
+        return new Promise((resolve) => {
+            const shell = new Powershell({});
+            shell.addCommand(`powershell -Command "Show-ControlPanelItem -Name '${searchResultItem.executionArgument}'"`)
+                .then(() => shell.invoke())
+                .catch((reason) => this.logger.error("Opening control panel item failed: " + reason))
+                .finally(() => shell.dispose());
+            resolve();
```

Comment:
I think you should resolve the promise only when the execution was successful. And on error you should reject the promise.

Reply:
I wrote about this in the initial comment:

> I intentionally don't await the powershell command to open a control panel item so that the ueli window is hidden right after pressing enter. Therefore I log errors that occur during execution right in the plugin.

I'm going to change this as you suggested because I think all the plugins should behave the same. But I don't like that the ueli window stays visible that long without any useful feedback if the exeuction takes some time. It feels like something didn't work... Maybe some execution indicator should be shown at least?

Reply:
Yes you're right, there should be an indicator that "something" is going on when pressing enter. But that's a separate task. Nevertheless, the promise here should only resolve on success and reject on error.